design-documents/bit-precise-types.rst

```diff
@@ -162,14 +420,17 @@ It has the following negatives:
 - The AArch64 ``LD{S,U}MAX`` operations would not work naturally on small values
   of this representation.
 
-- Operations ``/,%,==,<,>,<=,>=,>>`` and widening conversions would not
-  require extra work.
+- Operations ``/,%,==,<,>,<=,>=,>>`` and widening conversions on operands coming
+  from an ABI boundary would require masking the operands.
 
 - On AArch32 this could cause surprises to developers, given that on this
   architecture small Fundamental Data Types are have zero- or sign-extended
   extra bits.  So a ``char`` would not have the same representation as a
   ``_BitInt(8)`` on this architecture.
 
+- If used in calls to variadic functions which were written for standard
+  integral types this can give surprising results.
```

Comment:
Hmm, honestly I'd not seriously considered this since the way that the PCS is defined is through a two step process of
"source code type" -> "fundamental data type"
then
"fundamental data type" -> "rule for PCS passing".

Now you've pointed it out it does seem reasonable.
Not sure whether the exceptional case could lead to confusion -- will invite other feedback to see what people think.
@smithp35 @stuij @rsandifo-arm

Reply:
Test @rearnsha (wanted to tag you, the comment box doesn't want to auto-complete your name, don't know if typing it fully out will properly link -- this is a test).